libkbfs/block_journal.go

```diff
@@ -778,6 +847,55 @@ func flushBlockJournalEntry(
 	return nil
 }
 
+func flushBlockEntries(ctx context.Context, log logger.Logger,
+	bserver BlockServer, bcache BlockCache, reporter Reporter, tlfID TlfID,
+	tlfName CanonicalTlfName, entries blockEntriesToFlush) error {
+	if !entries.flushNeeded() {
+		// Avoid logging anything when there's nothing to flush.
+		return nil
+	}
+
+	// Do all the put state stuff first, in parallel.  We need to do
+	// the puts strictly before the addRefs, since the latter might
+	// reference the former.
+	log.CDebugf(ctx, "Putting %d blocks", len(entries.puts.blockStates))
+	blocksToRemove, err := doBlockPuts(ctx, bserver, bcache, reporter,
```

Comment:
Puts/addRefs to the block server are idempotent, so I don't think there should be any additional issues here.